presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -818,26 +769,17 @@ private PhysicalOperation createMergeSource(RemoteSourceNode node, LocalExecutio
 
         private PhysicalOperation createRemoteSource(RemoteSourceNode node, LocalExecutionPlanContext context)
         {
-            // Hack for now !@#$%^&*
-            context.setDriverInstanceCount(1);
-//            if (node.isEnsureSourceOrdering()) {
-//                context.setDriverInstanceCount(1);
-//            }
-//            else if (!context.getDriverInstanceCount().isPresent()) {
-//                context.setDriverInstanceCount(getTaskConcurrency(session));
-//            }
-
-            Map<PlanNodeId, Iterator<SerializedPage>> sparkShffuledSources = context.getSparkShuffledSources();
-
-            Optional<SparkExchangeSource> sparkExchangeSource = Optional.ofNullable(sparkShffuledSources.get(node.getId()))
-                    .map(iterator -> new SparkExchangeSource(iterator));
+            if (node.isEnsureSourceOrdering()) {
+                context.setDriverInstanceCount(1);
+            }
+            else if (!context.getDriverInstanceCount().isPresent()) {
+                context.setDriverInstanceCount(getTaskConcurrency(session));
```

Comment:
Do how do we enforce there is only one driver instance for Spark exchange client?

Reply:
Oh I see. Because `SparkRemoteSourceOperator` supports multiple threading.